modules/andes-core/broker/src/main/java/org/wso2/andes/store/file/FileMessageStoreImpl.java

```
@@ -0,0 +1,1364 @@
+
+                if (!keySplit[keySplit.length - 1].equals(EXPIRATION_TIME)) {
+                    break;
+                }
+                Long messageID = Long.parseLong(keySplit[2]);
+                Long messageExpiredTime = Long.parseLong(asString(keyIterator.peekNext().getValue()));
+
+                if (messageExpiredTime < currentTime) {
+                    expiredMessageIDList.add(messageID);
+                }
+                keyIterator.next();
+            }
+        } finally {
+            try {
+                keyIterator.close();
+            } catch (IOException e) {
+                log.error("Key iterator closing failed.", e);
+            }
+        }
+        return expiredMessageIDList;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Long> getExpiredMessagesFromDLC(long messageCount) throws AndesException {
+        List<Long> expiredMessageIDList = getExpiredMessages(0, DLC);
+        return expiredMessageIDList.subList(0, (int) messageCount);
+    }
+
+    @Override
+    public void addMessageToExpiryQueue(Long messageId, Long expirationTime, boolean isMessageForTopic, String
+            destination) throws AndesException {
+    }
+
+    @Override
+    public long getApproximateQueueMessageCount(String storageQueueName) throws AndesException {
+        return 0;
+    }
+
+    @Override
+    public void resetMessageCounterForQueue(String storageQueueName) throws AndesException {
+
+    }
+
+    @Override
+    public void removeLocalQueueData(String storageQueueName) {
+        // no cache is used
+    }
+
+    @Override
+    public void incrementMessageCountForQueue(String storageQueueName, long incrementBy) throws AndesException {
+
+    }
+
+    @Override
+    public void decrementMessageCountForQueue(String storageQueueName, long decrementBy) throws AndesException {
+
+    }
+
+    @Override
+    public void storeRetainedMessages(Map<String, AndesMessage> retainMap) throws AndesException {
```

Better to throw "NotImplementedException" in these methods. Otherwise they will silently fail.